### PR TITLE
Be 16

### PR DESCRIPTION
**Pull request by hananFr**
# JWT Service Implementation

This PR introduces a JWT service that handles token generation, validation, and verification. 

## Key Features:
- Token generation with RS256 algorithm
- Refresh token support with whitelist management
- Payload validation for required claims (sub, role)
- Comprehensive error handling
- Unit tests for core functionality
- MongoDB integration for whitelist management with TTL index

## Recent Updates:
- Migrated from blacklist to whitelist approach for better performance and maintainability
- Removed redundant functions and simplified the codebase
- Enhanced concurrent session handling
- Updated tests to match new implementation

## Technical Details:
- Uses RS256 algorithm for enhanced security
- Implements both access and refresh token flows
- Includes JTI (JWT ID) tracking in whitelist
- Comprehensive error handling with custom ApiError responses
- Full unit test coverage

## Note: 
Currently using file-based RSA keys for development. 
Production implementation will require:
- Decision on secrets storage solution (AWS Secrets Manager/Parameter Store/etc.)
- Implementation of secure key retrieval
- Update of configuration and initialization flow

## Testing:
All unit tests are passing with good coverage including error cases and edge conditions.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -79,5 +81,6 @@
     "uuid": "^11.0.5",
     "winston": "^3.11.0",
     "winston-elasticsearch": "^0.17.4"
-  }
+  },
+  "packageManager": "pnpm@9.15.4+sha512.b2dc20e2fc72b3e18848459b37359a32064663e5627a51e4c74b2c29dd8e8e0491483c3abb40789cfd578bf362fb6ba8261b05f0387d76792ed6e23ea3b1b6a0"
```
Comment (ShaySarussiElshten) [inline package.json:85]: i need this ?

File: src/config/index.ts [hunk 2]
```diff
@@ -13,12 +13,16 @@ const getConfig = (): IConfig => ({
   logLevel: process.env.LOG_LEVEL || 'info',
   corsOrigins: process.env.CORS_ORIGINS?.split(',') || ['*'],
   cookieSecret: process.env.COOKIE_SECRET || 'default-secret',
-  apiVersion: process.env.API_VERSION || 'v1'
+  apiVersion: process.env.API_VERSION || 'v1',
```
Comment (ShaySarussiElshten) [inline src/config/index.ts:16]: If you make changes to the .env file, please update instruction.md as well. Otherwise, I won't be able to execute your code successfully. :)

File: src/services/jwt/dbService.ts
Comment (ShaySarussiElshten) [inline src/services/jwt/dbService.ts:1]: if you not use in this file why you push this to your pr ?

**Review by ShaySarussiElshten — CHANGES_REQUESTED** [review]
Overall, the code looks great! Just make the changes I requested, and I'll continue with the review 💯 .

File: package.json (hunk 1 above)
Comment (hananFr) [reply package.json]: Removed the packageManager line as requested.

File: src/config/index.ts (hunk 2 above)
Comment (hananFr) [reply src/config/index.ts:16]: Updated instruction.md with RSA key generation steps and required .env configuration.

File: src/services/jwt/dbService.ts
Comment (hananFr) [reply src/services/jwt/dbService.ts:1]: Removed the unused dbService.ts file.

**Comment by hananFr** [comment]
I've made all the requested changes. Please let me know if anything else needs to be adjusted.

**Review by ShaySarussiElshten — APPROVED** [review]
